python_modules/dagster/dagster/core/storage/schedules/sql_schedule_storage.py

```diff
@@ -41,31 +40,43 @@ def execute(self, query):
     def _deserialize_rows(self, rows):
         return list(map(lambda r: deserialize_json_to_dagster_namedtuple(r[0]), rows))
 
-    def all_instigator_state(self, repository_origin_id=None, instigator_type=None):
+    def all_instigator_state(
+        self, repository_origin_id=None, repository_name=None, instigator_type=None
+    ):
         check.opt_inst_param(instigator_type, "instigator_type", InstigatorType)
-        base_query = db.select([JobTable.c.job_body, JobTable.c.job_origin_id]).select_from(
-            JobTable
-        )
 
-        if repository_origin_id:
-            query = base_query.where(JobTable.c.repository_origin_id == repository_origin_id)
+        if self.has_instigators_table() and self.has_built_index(SCHEDULE_JOBS_SELECTOR_ID):
+            query = db.select([InstigatorTable.c.instigator_body]).select_from(InstigatorTable)
+            if repository_name:
+                query = query.where(InstigatorTable.c.repository_name == repository_name)
```

Comment:
Ah I guess this answers my question on the earlier PR. It still seems a little random to me that this is the only filtering option, but I guess that's bc we happen to have a repository-scoped page